presto-orc/src/main/java/com/facebook/presto/orc/metadata/statistics/DecimalStatistics.java

```diff
@@ -25,14 +28,27 @@
     // 1 byte to denote if null
     public static final long DECIMAL_VALUE_BYTES_OVERHEAD = Byte.BYTES;
 
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(DecimalStatistics.class).instanceSize();
+    private static final int BIG_DECIMAL_INSTANCE_SIZE = ClassLayout.parseClass(BigDecimal.class).instanceSize() + ClassLayout.parseClass(BigInteger.class).instanceSize();
```

Comment:
`BigInteger` contains and `int[]` that has the bits. So you might want to add in the int array overhead (the body of the array is the same as `decimalSizeInBytes`)